alacritty_terminal/src/ansi.rs

```diff
@@ -24,63 +24,59 @@ use vte;
 
 use crate::term::color::Rgb;
 
+fn is_rgb_separator(ch: Option<char>) -> bool {
+    ch == None || ch == Some('/') || ch == Some('\x07')
+}
+
 // Parse color arguments
-//
-// Expect that color argument looks like "rgb:xx/xx/xx" or "#xxxxxx"
+// Expect the color argument in the form "rgb:x/x/x" or "#xxx", where x is 1-4 digit hex numbers
 fn parse_rgb_color(color: &[u8]) -> Option<Rgb> {
-    let mut iter = color.iter();
-
-    macro_rules! next {
-        () => {
-            iter.next().map(|v| *v as char)
-        };
+    let len = color.len();
+    // Legacy colors are in the form "#xxx"
+    // The use of this format is no longer encouraged by xparsecolor
+    let mut is_legacy = false;
+
+    if len >= 5 && color[0] == b'#' {
```

Comment:
We shouldn't account for the bell character here, instead it should be stripped before passing the string to `parse_rgb_color`. Otherwise it might cause problems when we want to reuse this in other places, since `4` is a perfectly valid length.